lib/Dialect/TritonGPU/Transforms/OptimizeDotOperands.cpp

```diff
@@ -110,10 +110,14 @@ class MoveOpAfterLayoutConversion : public mlir::RewritePattern {
       if (isa<triton::LoadOp>(currOp))
         checkOp = true;
       else if (checkOp) {
+        // Bail out if there exists an op after Load that is not FpToFp, Bitcast, or Arith.
         if (!isa<triton::FpToFpOp, triton::BitcastOp>(currOp) &&
             currOp->getDialect()->getTypeID() !=
                 mlir::TypeID::get<arith::ArithDialect>())
           return mlir::failure();
+        // Also bail out if there exists an op after Load that is Select.
+        if (isa<arith::SelectOp>(currOp))
```

Comment:
It doesn't seem like the right solution to me, since select is still a layout preserving op.

Reply:
@ThomasRaoux If the number of operands is >=2, should we still move forward the cvtOp? I'm concerning about the cost of extra cvts

Reply:
> It doesn't seem like the right solution to me, since select is still a layout preserving op.

Yes I'm don't see why select is different than other ops here? That seems arbitrary.

> @ThomasRaoux If the number of operands is >=2, should we still move forward the cvtOp? I'm concerning about the cost of extra cvts

It's hard to tell as in general the data should be constant or coming a pipelined load so there wouldn't necessarily be a cvt. If we want more complex heuristic we should probably deal with this as a more global heuristic but without real life examples it is hard to tell.

Reply:
It feels like this is just working around the problem that the code below assumes all the operand types are the same. I would prefer fixing this code even if we later want to restrict the propagation for heuristic reasons.

Reply:
Sounds good to me. Let's see if @manman-ren would like to take a stab on fixing the problem.

Reply:
Given that select is preserving the layout, the issue is when generating the ConvertLayoutOp, we use the same newCvtTy while select takes different operand types: i1 vs. another type. We may consider bailing out for select due to heuristic reasons, but for now we should fix the issue by using different newCvtTy for the i1 operand. I will try to fix it.

Reply:
@Jokeren @ThomasRaoux Added a commit on top of the stack, let me how it looks. Thanks!

Reply:
LGTM now

Reply:
Thanks @Jokeren! Also fixed a clang-format issue. It seems to indicate that "Merging can be performed automatically with 1 approving review.", I guess once it is approved, and there are no issues with ci, the PR will be merged. I don't think I have commit access as this is my first PR for Triton.